python/ray/train/batch_predictor.py

```diff
@@ -114,7 +132,15 @@ def __init__(self):
                 )
 
             def __call__(self, batch):
-                prediction_output = self.predictor.predict(batch, **predict_kwargs)
+                if feature_columns:
+                    prediction_batch = batch[feature_columns]
+                else:
+                    prediction_batch = batch
+                prediction_output = self.predictor.predict(
+                    prediction_batch, **predict_kwargs
+                )
+                if keep_columns:
+                    prediction_output[keep_columns] = batch[keep_columns]
```

Comment:
Shouldn't this also include the output column? It's weird if we drop the output column ever.